src/hgraph/nodes/_control_operators.py

```diff
@@ -0,0 +1,152 @@
+from collections import defaultdict
+from typing import Generic, cast, Type
+
+from hgraph import graph, all_, TSL, TS, SIZE, reduce, bit_and, any_, bit_or, compute_node, merge, TIME_SERIES_TYPE, \
+    REF, TSB, PythonTimeSeriesReference, TimeSeriesSchema, Size, AUTO_RESOLVE, STATE, EvaluationClock, MAX_DT, race
+
+__all__ = ("if_", "if_true", "if_then_else", "route_by_index", "BoolResult")
+
+
+@graph(overloads=all_)
+def all_default(*args: TSL[TS[bool], SIZE]) -> TS[bool]:
+    return reduce(bit_and, args, False)
+
+
+@graph(overloads=any_)
+def any_default(*args: TSL[TS[bool], SIZE]) -> TS[bool]:
+    """
+    Graph version of python `any` operator
+    """
+    return reduce(bit_or, args, False)
+
+
+@compute_node(overloads=merge)
+def merge_default(*tsl: TSL[TIME_SERIES_TYPE, SIZE]) -> TIME_SERIES_TYPE:
+    """
+    Selects and returns the first of the values that tick (are modified) in the list provided.
+    If more than one input is modified in the engine-cycle, it will return the first one that ticked in order of the
+    list.
+    """
+    return next(tsl.modified_values()).delta_value
+
+
+@compute_node(overloads=race)
+def race_default(*tsl: TSL[REF[TIME_SERIES_TYPE], SIZE],
```

Comment:
Where possible use schema based state to make this easier to support persistence in the future. This also helps reduce requirements for start functions where they only initialise state.